django/db/models/functions/comparison.py

```diff
@@ -159,35 +159,40 @@ def as_sql(self, compiler, connection, **extra_context):
             )
         return super().as_sql(compiler, connection, **extra_context)
 
-    def as_postgresql(self, compiler, connection, **extra_context):
-        copy = self.copy()
-        copy.set_source_expressions(
-            [
-                Cast(expression, TextField()) if index % 2 == 0 else expression
-                for index, expression in enumerate(copy.get_source_expressions())
-            ]
-        )
-        return super(JSONObject, copy).as_sql(
-            compiler,
-            connection,
-            function="JSONB_BUILD_OBJECT",
-            **extra_context,
-        )
-
-    def as_oracle(self, compiler, connection, **extra_context):
+    def as_native(self, compiler, connection, *, returning, **extra_context):
         class ArgJoiner:
             def join(self, args):
-                args = [" VALUE ".join(arg) for arg in zip(args[::2], args[1::2])]
-                return ", ".join(args)
+                pairs = zip(args[::2], args[1::2], strict=True)
+                return ", ".join([" VALUE ".join(pair) for pair in pairs])
```

Comment:
I'm late to the party, but something that I usually recommend is to avoid creating a whole list in memory before calling `join`, since it handles generators very nicely and it uses less memory:
```suggestion
                return ", ".join(" VALUE ".join(pair) for pair in pairs)
```

Reply:
A list comprehension is preferable here as `str.join()` converts to list internally anyway. It is better performance to provide a list up front.
 - https://stackoverflow.com/questions/9060653/list-comprehension-without-in-python/9061024#9061024
 - https://github.com/adamchainz/flake8-comprehensions/issues/156

Reply:
Thank you! Now that you mention it, I recall this was already discussed in another Django PR. I need to re-learn this! :books: